components/common/src/lib.rs

```diff
@@ -199,7 +199,7 @@ pub mod sync {
         let heartbeats = &THREAD_HEARTBEATS.lock()
                                            .expect("THREAD_HEARTBEATS poisoned");
         for (name, last_heartbeat) in threads_missing_heartbeat(heartbeats, threshold) {
-            error!("No heartbeat from {} in {} seconds; deadlock likely",
+            trace!("No heartbeat from {} in {} seconds; deadlock likely",
```

Comment:
I think this may be better at `warn`, rather than `trace`.

By default, `error` is the default level that is printed, but this really should be a higher level of urgency than `trace`.

Reply:
The problem is that warn will still spam when we have debug on, eg, acceptance

Reply:
FYI, you can configure different modules individually.

For instance:

```
RUST_LOG=debug,habitat_common::sync=error
```

will set everything to `DEBUG`, but set the level to `ERROR` for code coming from the `habitat_common::sync` module.

Reply:
(I don't think there's a way to selectively _disable_ logging, apart from setting the level, so something at the `error` level seems destined to always log. Moving it to `warn` seems like a good compromise between the "urgency"  of the message and the ability to control when it is output.)

Reply:
Ok, so changing it to warn and using the RUST_LOG config should work for now (we'll have to confirm after the change is merged that its working as expected).

Reply:
@chefsalim Turns out I was wrong; you _can_ selectively disable logging:

```
RUST_LOG=debug,habitat_common::sync=off
```
That's buried deep in the code of `env_logger`, but not clearly documented. With that, this PR wouldn't be necessary at all.

![tenor-73718303](https://user-images.githubusercontent.com/207178/58838074-8f747300-862b-11e9-9768-122c465255a7.gif)